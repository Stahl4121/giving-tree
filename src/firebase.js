//Reasoning for this file: https://stackoverflow.com/questions/48492047/where-do-i-initialize-firebase-app-in-react-application

// Firebase App (the core Firebase SDK) is always required and
// must be listed before other Firebase SDKs
import * as firebase from "firebase/app";

// Add the Firebase services that you want to use
import "firebase/auth";
import "firebase/firestore";

//Import everything, not recommended for production apps
//import firebase from "firebase";

const firebaseConfig = {
  apiKey: "AIzaSyAkeCK6w-qbvHzaOdEyg_PtpGQbElAv0_M",
  authDomain: "givingtree-cfs.firebaseapp.com",
  databaseURL: "https://givingtree-cfs.firebaseio.com",
  projectId: "givingtree-cfs",
  storageBucket: "givingtree-cfs.appspot.com",
  messagingSenderId: "709659064913",
  appId: "1:709659064913:web:d9decf22beb442f2e79670",
  measurementId: "G-KZ6PNG9K64"
};


firebase.initializeApp(firebaseConfig);

export default !firebase.apps.length ? firebase.initializeApp(firebaseConfig) : firebase.app();