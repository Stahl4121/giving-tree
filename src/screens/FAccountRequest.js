import React from 'react';

import Text from '../components/Text.js';

import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  marginBottom: {
    marginBottom: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  links: {
    textDecoration: 'none'
  },
  errorMsg: {
    color: 'red',
    marginTop: theme.spacing(1),
  },
}));


function FAccountRequest() {
  const classes = useStyles();

  return (
    <Container component="main" maxWidth="md" >
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Text type='card-heading' text={'Foundation Account Request'} />
        <form className={classes.form} noValidate>
          <Grid container spacing={2}>

          <Container maxWidth="xs" className={classes.marginBottom}>
            <Grid container spacing={2}>
              <Grid container item xs={12}>
                <Text type='card-sectionheading' text={'Foundation Info'} />
                <Text type='card-subheading' text={'This information will be displayed publicly on all published giving opportunities.'} />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  name="name"
                  variant="outlined"
                  required
                  fullWidth
                  id="name"
                  label="Foundation Name"
                  autoFocus
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  name="foundation_url"
                  variant="outlined"
                  required
                  fullWidth
                  id="foundation_url"
                  label="Website URL"
                  placeholder="www.communityfoundation.com"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="public_email"
                  variant="outlined"
                  required
                  fullWidth
                  id="public_email"
                  label="Public Email"
                  placeholder="contact@cf.com"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="public_phone"
                  variant="outlined"
                  required
                  fullWidth
                  id="public_phone"
                  label="Public Phone"
                />
              </Grid>
            </Grid>
            </Container>

            <Container maxWidth="xs">
            <Grid container spacing={2}>
              <Grid container item xs={12}>
                <Text type='card-sectionheading' text={'Personal Contact Info'} />
                <Text type='card-subheading' text={'This information will be used to sign into and manage your account.'} />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="fname_contact"
                  variant="outlined"
                  required
                  fullWidth
                  id="fname_contact"
                  label="First Name"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="lname_contact"
                  label="Last Name"
                  name="lname_contact"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="personal_email"
                  label="Personal Email"
                  name="personal_email"
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="personal_phone"
                  variant="outlined"
                  required
                  fullWidth
                  id="personal_phone"
                  label="Personal Phone"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="passwordOne"
                  label="Password"
                  type="password"
                  id="passwordOne"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="passwordTwo"
                  label="Confirm Password"
                  type="password"
                  id="passwordTwo"
                />
              </Grid>
            </Grid>
            </Container>

            
            <Container maxWidth="xs">
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Request Account
          </Button>
            <Grid container justify="flex-end">
              <Grid item>
                <Link href="/signin" variant="body2">
                  Already have an account? Sign in
              </Link>
              </Grid>
            </Grid>
            </Container>

          </Grid>
        </form>
      </div>
    </Container>
  );
}

export default FAccountRequest;