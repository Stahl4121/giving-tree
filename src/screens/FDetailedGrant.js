import React, {useEffect} from 'react';
export default function FDetailedGrant() {
  // Set tab title
  useEffect(() => { document.title = 'Grant name here' }, []);

  return(
    <p>Foundation Detailed Grant!</p>
  );
}