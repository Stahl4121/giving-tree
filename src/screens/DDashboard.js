import React, { useEffect } from 'react';
import SmallGrantCard from '../components/SmallGrantCard.js';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { useCollection } from 'react-firebase-hooks/firestore';
import firebase from '../firebase.js';
import Search from '../components/Search';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  container: {
    paddingBottom: theme.spacing(2),
  },
}));

export default function DDashboard() {
  // List of small grant cards
  const [grants, setGrants] = React.useState([]);

  // Initialize database and specific grant in database
  const db = firebase.firestore();

  const [snapshot, loading, error] = useCollection(db.collection('grants'));

  useEffect(() => {
    var newGrants = [];
    if (!loading && !error) {
      snapshot.forEach(function (doc) {
        newGrants.push(
          <SmallGrantCard
            id={doc.id}
            title={doc.data().title}
            cfName={doc.data().cf_name}
            nonprofitName={doc.data().nonprofit_name}
            goalAmt={doc.data().goal_amt}
            moneyRaised={doc.data().money_raised}
            img={doc.data().images[0] || 'GivingTree.png'} />
        );
      });
      setGrants(newGrants);
    }
  }, [snapshot, error, loading]);
  /*
  useEffect(() => {
    var useless = 0;
  }, [grants]);
  */
  function searchCallback(childData) {
    var newGrants = [];
    childData.forEach((grant) => {
      newGrants.push(
        <SmallGrantCard
          id={grant.id}
          title={grant.title}
          cfName={grant.cf_name}
          nonprofitName={grant.nonprofit_name}
          goalAmt={grant.goal_amt}
          moneyRaised={grant.money_raised}
          img={grant.images[0] || 'GivingTree.png'} />
      );
    });
    setGrants(newGrants);
  }

  const classes = useStyles();

  return (
    <Container maxWidth='md' className={classes.container}>
      <Search parentCallback={searchCallback} />
      <Grid container spacing={2} >
        {grants}
      </Grid>
    </Container>
  );
}
