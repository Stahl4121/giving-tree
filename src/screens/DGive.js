// https://stripe.com/docs/recipes/elements-react

import React, { useEffect } from 'react';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/styles';
import Text from '../components/Text.js';
import firebase from '../firebase.js';
import Snack from '../components/Snack.js';
import {
  CardElement,
  injectStripe
} from 'react-stripe-elements';
import Button from '@material-ui/core/Button';

//const FieldValue = require('firebase-admin').firestore.FieldValue;

const useStyles = makeStyles(theme => ({
  pageLayout: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
      width: 600,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2),
  },
  paymentPaper: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
    padding: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
      marginTop: theme.spacing(6),
      marginBottom: theme.spacing(6),
      padding: theme.spacing(3),
    },
  },
  stripeElement: {
    display: 'block',
    margin: '10px 0 20px 0',
    width: '100%',
    padding: '10px 14px',
    fontSize: '1em',
    fontFamily: 'Roboto',
    boxShadow: 'rgba(50, 50, 93, 0.14902) 0px 1px 3px, rgba(0, 0, 0, 0.0196078) 0px 1px 0px',
    border: '0',
    outline: '0',
    borderRadius: '4px',
    background: 'white',
  },
}))


function Stripe(props) {
  const classes = useStyles();

  const db = firebase.firestore();

  const [complete, setComplete] = React.useState(false);
  const [amount, setAmount] = React.useState('');
  const [grantName] = React.useState('Test Grant'); //React.useState(props.grantName);
  const [given, setGiven] = React.useState('');
  const [goal, setGoal] = React.useState('');
  const [grantID] = React.useState('wJArP9RCeQY9vDvsfIA2') //React.useState(props.grantID);

  const docRef = db.collection('grants').doc(grantID);


  useEffect(() => {
    docRef.onSnapshot((doc) => {
      if (doc.exists) {
        setGoal(doc.data().goal_amt);
        setGiven(doc.data().money_raised);
      } else {
        console.log('No such grant!');
      }
    })
  });

  const submit = async (ev) => {
    let { token } = await props.stripe.createToken({ name: 'Giving Tree Donor' });
    let response = await fetch('/charge', {
      method: 'POST',
      headers: { 'Content-Type': 'text/plain' },
      body: token.id + ' amount: ' + (amount * 100) + ' description: ' + grantName,
    });

    if (response.ok) {
      // docRef.update({
      //   money_raised: firebase.firestore.FieldValue.increment(amount)
      // });
      db.runTransaction(function (transaction) {
        // This code may get re-run multiple times if there are conflicts.
        return transaction.get(docRef).then(function (doc) {
          if (!doc.exists) {
            throw new Error("Document does not exist!");
          }

          // Note: this could be done without a transaction by updating the population using FieldValue.increment()
          var newMoneyRaised = doc.data().money_raised + Number.parseInt(amount);
          transaction.update(docRef, { money_raised: newMoneyRaised });
        });
      }).then(function () {
        setComplete(true);
      }).catch(function (error) {
        console.log("Transaction failed: ", error);
      });
    }
  }

  return (
    <Container className={classes.pageLayout}>
      <React.Fragment>
        <Paper elevation={3} className={classes.paymentPaper}>
          <Text type='card-heading' text={grantName} />
          {complete ?
            <Text type='card-subheading' text={'Thank you for your donation! Thanks to your gift of $' + amount + ', ' + grantName + ' is now only  $' + (goal - given) + ' from meeting its goal of $' + goal + '!'} />
            :
            <div>
              <Text type='card-subheading' text={'So far, $' + given + ' has been raised out of $' + goal} />
              <CardElement className={classes.stripeElement} />
              <input
                className={classes.stripeElement}
                placeholder="Amount"
                onInput={e => setAmount(e.target.value)} />
              <Button
                color="primary"
                className={classes.button}
                variant="contained"
                onClick={submit}>
                Donate
              </Button>
            </div>
          }
        </Paper>
      </React.Fragment>
    </Container>
  );
}

export default injectStripe(Stripe);