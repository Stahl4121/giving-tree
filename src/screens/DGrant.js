import React, { useEffect } from 'react';
import LargeGrantCard from '../components/LargeGrantCard.js';
import firebase from '../firebase.js';


export default function DGrant(props) {
  let grantId = props.location.state.grantId;

  let [grant, setGrant] = React.useState('');

  // Initialize database and specific grant in database
  const db = firebase.firestore();

  useEffect(() => {
    db.collection('grants').get(grantId).then((doc) => {
      // doc.data() is never undefined for query doc snapshots
      setGrant(<LargeGrantCard
        grant={doc.data().title}
        foundation={doc.data().cf_name}
        nonprofit={doc.data().nonprofit_name}
        goal={doc.data().goal_amt}
        raised={doc.data().money_raised}
        img={doc.data().images[0] || 'GivingTree.png'} />);
    })
    return function cleanup() {
      // TODO: unsubscribe from Firebase query
    }
  });

  return (
    <div>
      {grant}
    </div>
  );
}
