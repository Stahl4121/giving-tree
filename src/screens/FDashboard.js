import React, { useEffect, useContext } from 'react';
import SmallGrantCard from '../components/SmallGrantCard.js';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { useCollection } from 'react-firebase-hooks/firestore';
import firebase from '../firebase.js';
import Search from '../components/Search';
import { makeStyles } from '@material-ui/core/styles';
import UserAuthContext from '../auth/context.js';
import ToggleButton from '@material-ui/lab/ToggleButton';
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';

const useStyles = makeStyles(theme => ({
  container: {
    paddingBottom: theme.spacing(2),
  },
  toggleBar: {
    marginBottom: 4,
  }
}));

export default function FDashboard(props) {

  const classes = useStyles();

  // List of small grant cards
  const [grants, setGrants] = React.useState([]);
  const [docs, setDocs] = React.useState();
  const [expiredGrants, setExpiredGrants] = React.useState([]);
  const [currentGrants, setCurrentGrants] = React.useState([]);
  const [draftedGrants, setDraftedGrants] = React.useState([]);
  const [toggleBarStatus, setToggleBarStatus] = React.useState('current');

  // Foundation ID
  const user = useContext(UserAuthContext);

  // Initialize database and specific grant in database
  const db = firebase.firestore();
  const [snapshot, loading, error] = useCollection(db.collection('grants'));

  useEffect(() => {
    let newDocs = [];
    let curGrants = [];
    let exGrants = [];
    let drGrants = [];
    if (!loading && !error) {
      snapshot.forEach(function (doc) {
        if (doc.data().cf_id === user?.cfId) {
          newDocs.push({
            dist: -1,
            id: doc.id,
            title: doc.data().title,
            cfName: doc.data().cf_name,
            nonprofitName: doc.data().nonprofit_name,
            goalAmt: doc.data().goal_amt,
            moneyRaised: doc.data().money_raised,
            img: doc.data().images[0] || 'GivingTree.png',
            nonprofitId: doc.data().nonprofit_id,
            address: doc.data().address,
            lat: doc.data().lat,
            long: doc.data().long,
            datePosted: doc.data().date_posted,
            dateDeadline: doc.data().date_deadline,
            desc: doc.data().desc,
            tags: doc.data().tags,
            status: doc.data().status
          });

          let grant = <SmallGrantCard
            id={doc.id}
            title={doc.data().title}
            cfName={doc.data().cf_name}
            nonprofitName={doc.data().nonprofit_name}
            goalAmt={doc.data().goal_amt}
            moneyRaised={doc.data().money_raised}
            img={doc.data().images[0] || 'GivingTree.png'}
            status={doc.data().status} />;
          if (doc.data().status === 'current') {
            curGrants.push(grant);
          } else if (doc.data().status === 'draft') {
            drGrants.push(grant);
          } else if (doc.data().status === 'expired') {
            exGrants.push(grant);
          }
        }
      });
      setDocs(newDocs);
      setCurrentGrants(curGrants);
      setDraftedGrants(drGrants);
      setExpiredGrants(exGrants);
    }
  }, [snapshot, error, loading]);

  function searchCallback(childData) {
    // console.log("childData in dashboard: ", childData);
    var curGrants = [];
    var exGrants = [];
    var drGrants = [];
    childData.forEach((meta) => {
      let grant = <SmallGrantCard
        id={meta.grant.id}
        title={meta.grant.title}
        cfName={meta.grant.cfName}
        nonprofitName={meta.grant.nonprofitName}
        goalAmt={meta.grant.goalAmt}
        moneyRaised={meta.grant.moneyRaised}
        img={meta.grant.img} />
      if (meta.grant.status === 'current') {
        curGrants.push(grant);
      } else if (meta.grant.status === 'draft') {
        drGrants.push(grant);
      } else if (meta.grant.status === 'expired') {
        exGrants.push(grant);
      }
    });
    console.log("CURRENT GRANTS: " + curGrants);

    setCurrentGrants(curGrants);
    setDraftedGrants(drGrants);
    setExpiredGrants(exGrants);
  }

  const handleToggle = (event, status) => {
    setToggleBarStatus(status);
  };

  const buttonOptions = [
    <ToggleButton key={1} value="current">
      Current
    </ToggleButton>,
    <ToggleButton key={2} value="drafted">
      Drafted
    </ToggleButton>,
    <ToggleButton key={3} value="expired">
      Expired
    </ToggleButton>
  ];

  return (
    <Container maxWidth='md' className={classes.container}>
      {docs &&
        <div>
          <Grid container spacing={2} direction="column" alignItems="center">
            <Grid item className={classes.toggleBar}>
              <ToggleButtonGroup size="small" value={toggleBarStatus} exclusive onChange={handleToggle}>
                {buttonOptions}
              </ToggleButtonGroup>
            </Grid>
          </Grid>
          <Search docs={docs} parentCallback={searchCallback} />
          <Grid container spacing={2} >
            {toggleBarStatus === 'current' ? currentGrants : toggleBarStatus === 'drafted' ? draftedGrants : expiredGrants}
          </Grid>
        </div>
      }
    </Container>
  );
}
