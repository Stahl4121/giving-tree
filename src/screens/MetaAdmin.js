import React, { useEffect, useState } from 'react';
import { useCollection } from 'react-firebase-hooks/firestore';
import $ from 'jquery';

import FoundationCard from '../components/FoundationCard.js';
import firebase from '../firebase.js';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
import ToggleButton from '@material-ui/lab/ToggleButton';
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';

const useStyles = makeStyles(theme => ({
  container: {
    paddingBottom: theme.spacing(2),
  },
  toggleBar: {
    marginBottom: 4,
  }
}));

function MetaAdmin(props) {
  const classes = useStyles();

  const [toggleBarStatus, setToggleBarStatus] = React.useState('current');
  const [reqCfs, setReqCfs] = React.useState();
  const [curCfs, setCurCfs] = React.useState();
  const [denCfs, setDenCfs] = React.useState();

  // Initialize database
  const db = firebase.firestore();
  const [snapshot, loading, error] = useCollection(db.collection('communityFoundations'));

  useEffect(() => {
    let req = [];
    let cur = [];
    let den = [];

    if (!loading && !error) {
      snapshot.forEach((doc) => {
        const card = <FoundationCard data={doc.data()} approveCB={approveCB} denyCB={denyCB} deleteCB={deleteCB} />;

        switch (doc.data().status) {
          case 'requested':
            req.push(card);
            break;
          case 'current':
          case 'deactivated':
            cur.push(card);
            break;
          case 'denied':
            den.push(card);
            break;
          default:
            console.log("Error, Bad CF Document: " + doc.id + ", " + doc.data());
        }
      })

      setReqCfs(req);
      setCurCfs(cur);
      setDenCfs(den);
    }
  }, [snapshot, error, loading]);

  const approveCB = (email) => { updateCf(email, 'current'); }
  const denyCB = (email) => { updateCf(email, 'denied'); }
  const deleteCB = (email) => { updateCf(email, 'deleted'); }

  const updateCf = (email, status) => {

    // User is signed in. Get the ID token.
    firebase.auth().currentUser.getIdToken(true)
      .then((idToken) => {
        // Pass the ID token to the server.
        $.post(
          '/updateCf',
          {
            idToken: idToken,
            cfEmail: email,
            cfStatus: status,
          },
          (data, postStatus) => {
            const json = JSON.parse(data);

            if (postStatus === 'success' && data && json && json.status === 'success') {

              db.collection("communityFoundations").where("personal_email", "==", email)
                .get()
                .then((querySnapshot) => {
                  querySnapshot.forEach((doc) => {

                    if (status === 'deleted') {
                      db.collection("communityFoundations").doc(doc.id).delete()
                        .then(() => {
                          console.log("Document successfully deleted!");
                        })
                        .catch((error) => {
                          console.error("Error deleting CF document: ", error);
                        });
                    }
                    else {
                      const time = firebase.firestore.Timestamp.now();
                      const dateField = (status === 'current') ? { date_approved: time } :
                        ((status === 'denied') ? { date_denied: time } : {});

                      db.collection("communityFoundations").doc(doc.id).update({ ...dateField, status: status })
                        .then(() => {
                          console.log("Document successfully updated!");
                        })
                        .catch((error) => {
                          console.error("Error updating CF document: ", error);
                        });
                    }

                  });
                })
                .catch((error) => {
                  console.error("Error retrieving CF document: ", error);
                });

            }
            else {
              console.error("Error with custom claims post status: " + error);
            }

          })
          .catch((error) => {
            console.error("Error with custom claims post: " + error);
          });
      })
      .catch((error) => {
        console.error("Error with getIdToken(): " + error);
      });
  }

  const handleToggle = (event, status) => {
    setToggleBarStatus(status);
  };

  const buttonOptions = [
    <ToggleButton key={1} value="requested">
      Requested
    </ToggleButton>,
    <ToggleButton key={2} value="current">
      Current
    </ToggleButton>,
    <ToggleButton key={3} value="denied">
      Denied
    </ToggleButton>
  ];

  return (
    <Container maxWidth='md' className={classes.container}>
      <Typography variant="h4">Meta-Admin</Typography>
      <Grid container spacing={2} direction="column" alignItems="center">
        <Grid item className={classes.toggleBar}>
          <ToggleButtonGroup size="small" value={toggleBarStatus} exclusive onChange={handleToggle}>
            {buttonOptions}
          </ToggleButtonGroup>
        </Grid>
      </Grid>
      <Grid item spacing={2} xs={12} >
        {toggleBarStatus === 'requested' ? reqCfs : toggleBarStatus === 'current' ? curCfs : denCfs}
      </Grid>
    </Container>
  );
}

export default MetaAdmin;