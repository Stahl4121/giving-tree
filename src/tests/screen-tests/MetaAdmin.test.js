import React from 'react';
import { shallow } from 'enzyme';
import FirestoreMock from '../firestore.mock'
import { MetaAdmin } from '../../screens/MetaAdmin';
import firebase from 'firebase/app';
import 'firebase/firestore';

const mockCfs = [
  {
    name: 'The Test Foundation',
    public_email: 'testy@tester.tested',
    public_phone: '1112223333',
    foundation_url: 'testme.com',
    fname_contact: 'Tabaath',
    lname_contact: 'Ben Test',
    personal_email: 'blah@yada.net',
    personal_phone: '9998887777',
    status: 'current',
    date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
    date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
    date_denied: '',
    date_deactivated: '',
  },
  {
    name: 'Charity Foundation',
    public_email: 'charity@mail.com',
    public_phone: '12345678945',
    foundation_url: 'charity.com',
    fname_contact: 'Chareth',
    lname_contact: 'Tyler',
    personal_email: 'chareth@yahoo.com',
    personal_phone: '4561237894',
    status: 'requested',
    date_requested: 'March 09, 2020 at 9:09:34 AM UTC-4',
    date_approved: '',
    date_denied: '',
    date_deactivated: '',
  },
]

describe('MetaAdmin', () => {

  beforeEach(() => {
    // wrapper = shallow(<MetaAdmin />);
  });

  // it('renders shallowly without crashing', () => {
  //   const wrapper = shallow(<MetaAdmin />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  describe('uses firestore', () => {
    const firestoreMock = new FirestoreMock();

    beforeEach(() => {
      firebase.firestore().collection = firestoreMock.mockCollection;
      firestoreMock.reset();
    });

    it('call community foundations snapshot on render', async () => {
      firestoreMock.mockOnSnapshotSuccess = mockCfs;

      const wrapper = shallow(<MetaAdmin />);
      wrapper.update();

      expect(firestoreMock.mockCollection).toBeCalledWith('communityFoundations');
      // expect(firestoreMock.mockOnSnapshot).toBeCalled(1);
    });
  });

});
// describe('The Agreement model', () => {
//   const firestoreMock = new FirestoreMock()
//   beforeEach(() => {
//       firebase.firestore = firestoreMock
//       firestoreMock.reset()
//   })

//   it('does something', (done) => {
//       firestoreMock.mockAddReturn = { id: 'test-id' }
//       firebase.firestore.collection('foobar')
//         .add({foo: 'bar'})
//         .then(res => {
//           expect(firestoreMock.mockCollection).toBeCalledWith('foobar')
//           expect(firestoreMock.mockAdd).toBeCalledWith({foo: 'bar'})
//           expect(res.id).toEqual('test-id')
//           done()
//         })
//         .catch(done)
//   })
// })