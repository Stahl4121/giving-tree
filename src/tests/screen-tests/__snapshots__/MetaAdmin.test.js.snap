// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaAdmin renders shallowly without crashing 1`] = `ShallowWrapper {}`;
